hpx/parallel/util/numa_binding_allocator.hpp

```diff
@@ -0,0 +1,639 @@
+          , policy_(rhs.policy_)
+          , flags_(rhs.flags_)
+        {
+            LOG_NUMA_MSG("numa_binding_allocator : Move constructor");
+        }
+
+        // Assignment operator
+        numa_binding_allocator& operator=(numa_binding_allocator const& rhs)
+        {
+            binding_helper_ = rhs.binding_helper_;
+            policy_ = rhs.policy_;
+            flags_ = rhs.flags_;
+
+            LOG_NUMA_MSG("numa_binding_allocator : Assignment operator");
+            return *this;
+        }
+
+        // Move assignment
+        numa_binding_allocator& operator=(numa_binding_allocator&& rhs)
+        {
+            binding_helper_ = rhs.binding_helper_;
+            policy_ = rhs.policy_;
+            flags_ = rhs.flags_;
+
+            LOG_NUMA_MSG("numa_binding_allocator : Move assignment");
+            return *this;
+        }
+
+        // Returns the actual address of x even in presence of overloaded
+        // operator&
+        pointer address(reference x) const noexcept
+        {
+            return &x;
+        }
+
+        // Returns the actual address of x even in presence of overloaded
+        // operator&
+        const_pointer address(const_reference x) const noexcept
+        {
+            return &x;
+        }
+
+        // Allocates n * sizeof(T) bytes of uninitialized storage and
+        // then spawns threads to touch memory if membind_user is selected
+        pointer allocate(size_type n)
+        {
+            pointer result = nullptr;
+
+            if (policy_ ==
+                threads::hpx_hwloc_membind_policy::membind_firsttouch)
+            {
+                threads::hwloc_bitmap_ptr bitmap =
+                    threads::get_thread_manager().get_pool_numa_bitmap(
+                        binding_helper_->pool_name());
+                //
+                result = reinterpret_cast<pointer>(
+                    threads::topology().allocate_membind(
+                        n * sizeof(T), bitmap, policy_, 0));
+            }
+            else if (policy_ ==
+                threads::hpx_hwloc_membind_policy::membind_interleave)
+            {
+                threads::hwloc_bitmap_ptr bitmap =
+                    threads::get_thread_manager().get_pool_numa_bitmap(
+                        binding_helper_->pool_name());
+                //
+                result = reinterpret_cast<pointer>(
+                    threads::topology().allocate_membind(
+                        n * sizeof(T), bitmap, policy_, 0));
+            }
+            else if (policy_ == threads::hpx_hwloc_membind_policy::membind_user)
+            {
+                result = reinterpret_cast<pointer>(
+                    threads::topology().allocate(n * sizeof(T)));
+                threads::hwloc_bitmap_ptr bitmap =
+                    threads::get_thread_manager().get_pool_numa_bitmap(
+                        binding_helper_->pool_name());
+                //
+                result = reinterpret_cast<pointer>(
```

Comment:
@biddisco not sure if you're aware, but `result` is still being assigned to twice here.